### PR TITLE
RobotFactoryPosition Layer

### PR DESCRIPTION
**Pull request by p-nayak11**
## Description
This PR is a refactor of the strategy flow to remove the Coach Node and replace it with the RobotFactoryPosition. This refactor's motivations are explained in the this [design document](https://docs.google.com/document/d/1eVusKIoqb7wYXdVYjSSHk3PGomOQd3RMymBeRynhVuU/edit?usp=sharing).  

Some other notes (Sid):
- this PR deletes the long broken Position control UI. will think about a replacement, but it is noticeably harder in the new paradigm. 

## Associated / Resolved Issue
[ClickUp card 1](https://app.clickup.com/t/863h9ct6x)
[ClickUp card 2](https://app.clickup.com/t/863h9ctf1)

## Design Documents
[Link](https://docs.google.com/document/d/1eVusKIoqb7wYXdVYjSSHk3PGomOQd3RMymBeRynhVuU/edit?usp=sharing)

### Test Case 1
1. Ensure that the coach node is not being spun
2. make run-sim

**Expected result:** 4 Defense, 1 Offense, 1 Goalie play shoot only. 

## Key Files to Review
Group 1
 * strategy/agent/position/robot_factory_position.cpp
 * strategy/agent/position/position.hpp

Group 2
 * planning/planner_node.cpp

## Review Checklist

- [x] **Docstrings**: All methods and classes should have the file appropriate docstrings which follow the guidelines in the ["Contributing" page](https://rj-rc-software.readthedocs.io/en/latest/contributing.html) of our docs.
- [x] **Remove extra print statements**: Any print statements used for debugging should be removed
- [ ] **Tag reviewers**: Tag some people for review and ping them on Slack

### CODE REVIEW

File: rj_common/include/rj_common/network.hpp
Comment (rishiso) [inline rj_common/include/rj_common/network.hpp:1]: Is this supposed to be committed?

Comment (sid-parikh) [reply rj_common/include/rj_common/network.hpp:1]: yes this is fine

File: rj_constants/include/rj_constants/topic_names.hpp [hunk 1]
```diff
@@ -69,7 +69,6 @@ static inline std::string trajectory_topic(int robot_id) {
 }  // namespace planning::topics
 
 namespace strategy::topics {
-constexpr auto kCoachStateTopic{"/strategy/coach_state"};
 constexpr auto kPositionsTopic{"/strategy/positions"};
```
Comment (sid-parikh) [inline rj_constants/include/rj_constants/topic_names.hpp]: delete this too (my bad)

**Comment by sid-parikh** [comment]
GH Action for style appears to be broken due to some issue related to either Docker or ROS's package hosting. I just pushed a commit that reformatted, so I'm ignoring the fix-style check.

**Review by sid-parikh — APPROVED** [review]
Congratulations @p-nayak11

**Pull request merged**